webdriver/tests/bidi/network/provide_response/invalid.py

```diff
@@ -0,0 +1,103 @@
+    await setup_network_test(events=[
+        "network.responseCompleted",
+    ])
+    on_response_completed = wait_for_event("network.responseCompleted")
+
+    text_url = url(PAGE_EMPTY_TEXT)
+    await fetch(text_url)
+
+    response_completed_event = await on_response_completed
+    request = response_completed_event["request"]["request"]
+
+    with pytest.raises(error.NoSuchRequestException):
+        await bidi_session.network.provide_response(request=request)
+
+
+@pytest.mark.parametrize("value", [False, 42, {}, []])
+async def test_params_reason_phrase_invalid_type(bidi_session,
+                                                 setup_network_test, url,
+                                                 fetch, wait_for_event,
+                                                 add_intercept, value):
+    request = await setup_blocked_request_test(setup_network_test, url,
+                                               add_intercept, fetch,
+                                               wait_for_event)
+
+    with pytest.raises(error.InvalidArgumentException):
+        await bidi_session.network.provide_response(request=request,
+                                                    reason_phrase=value)
+
+
+@pytest.mark.parametrize("value", [False, "s", {}, []])
+async def test_params_status_code_invalid_type(bidi_session,
+                                               setup_network_test, url, fetch,
+                                               wait_for_event, add_intercept,
+                                               value):
+    request = await setup_blocked_request_test(setup_network_test, url,
+                                               add_intercept, fetch,
+                                               wait_for_event)
+
+    with pytest.raises(error.InvalidArgumentException):
+        await bidi_session.network.provide_response(request=request,
+                                                    status_code=value)
+
+
+@pytest.mark.parametrize("value", [-1, 4.3])
+async def test_params_status_code_invalid_value(bidi_session,
+                                                setup_network_test, url, fetch,
+                                                wait_for_event, add_intercept,
+                                                value):
+    request = await setup_blocked_request_test(setup_network_test, url,
+                                               add_intercept, fetch,
+                                               wait_for_event)
+
+    with pytest.raises(error.InvalidArgumentException):
+        await bidi_session.network.provide_response(request=request,
+                                                    status_code=value)
+
+
+# TODO: Add cookies.
+# TODO: Add credentials.
+# TODO: Add headers.
+
+
+async def setup_blocked_request_test(setup_network_test, url, add_intercept,
+                                     fetch, wait_for_event):
+    await setup_network_test(events=["network.beforeRequestSent"])
```

Comment:
should be `network.responseStarted`.

Maybe we should really make that a shared helper which can handle different pairs of event/phase, this would avoid potential mistakes here.